src/handlers/swap_quote_handlers.ts

```diff
@@ -0,0 +1,199 @@
+import { ExtensionContractType, Orderbook, SwapQuoteConsumer, SwapQuoter, SwapQuoterError } from '@0x/asset-swapper';
+import { SupportedProvider } from '@0x/order-utils';
+import { BigNumber, RevertError } from '@0x/utils';
+import { TxData, Web3Wrapper } from '@0x/web3-wrapper';
+import * as express from 'express';
+import * as HttpStatus from 'http-status-codes';
+
+import { CHAIN_ID } from '../config';
+import {
+    DEFAULT_QUOTE_SLIPPAGE_PERCENTAGE,
+    DEFAULT_TOKEN_DECIMALS,
+    ETH_SYMBOL,
+    QUOTE_ORDER_EXPIRATION_BUFFER_MS,
+} from '../constants';
+import { InternalServerError, RevertAPIError, ValidationError, ValidationErrorCodes } from '../errors';
+import { isAPIError } from '../middleware/error_handling';
+import { GetSwapQuoteRequestParams, GetSwapQuoteResponse } from '../types';
+import { findTokenAddress, findTokenDecimalsIfExists } from '../utils/token_metadata_utils';
+
+export class SwapQuoteHandlers {
+    private readonly _provider: SupportedProvider;
+    private readonly _swapQuoter: SwapQuoter;
+    private readonly _swapQuoteConsumer: SwapQuoteConsumer;
+    private readonly _web3Wrapper: Web3Wrapper;
+    constructor(orderbook: Orderbook, provider: SupportedProvider) {
+        this._provider = provider;
+        const swapQuoterOpts = {
+            chainId: CHAIN_ID,
+            expiryBufferMs: QUOTE_ORDER_EXPIRATION_BUFFER_MS,
+        };
+        this._swapQuoter = new SwapQuoter(this._provider, orderbook, swapQuoterOpts);
+        this._swapQuoteConsumer = new SwapQuoteConsumer(this._provider);
+        this._web3Wrapper = new Web3Wrapper(this._provider);
+    }
+    public async calculateSwapQuoteAsync(
+        buyTokenAddress: string,
+        sellTokenAddress: string,
+        buyAmount: BigNumber | undefined,
+        sellAmount: BigNumber | undefined,
+        from: string | undefined,
+        isETHSell: boolean = false,
+        slippagePercentage?: number,
+        providedGasPrice?: BigNumber,
```

Comment:
nit: greater than 2 options => options object. Maybe for all these `| undefined` types too.